arelle/ValidateXbrlCalcs.py

```diff
@@ -278,6 +338,43 @@ def bindFacts(self, facts, ancestors):
                 if concept in self.conceptsInRequiresElement:
                     self.requiresElementFacts[concept].append(f)
 
+    def oimConsistentInterval(self, fList):
+        if any(f.isNil for f in fList):
+            if all(f.isNil for f in fList):
+                return (NIL_FACT_SET,NIL_FACT_SET)
+            _inConsistent = True # provide error message 
+        else: # not all have same decimals
+            f0 = fList[0]
+            _d = inferredDecimals(f0)
+            _v = f0.xValue
+            _inConsistent = isnan(_v) # NaN is incomparable, always makes dups inconsistent
+            decVals = {_d: _v}
+            aMax, bMin = rangeValue(_v, _d)
+            for f in fList[1:]:
+                _d = inferredDecimals(f)
+                _v = f.xValue
+                if isnan(_v):
+                    _inConsistent = True
+                    break
+                if _d in decVals:
+                    _inConsistent |= _v != decVals[_d]
+                else:
+                    decVals[_d] = _v
+                a, b = rangeValue(_v, _d)
+                if a > aMax: aMax = a
```

Comment:
Nit suggest: `aMax = max(aMax, a)`, same for `bMin = min(bMin, b)`

Reply:
Prefer inline code here since it's an inner loop the function overhead would be greater than inline (I think, unless python optimizes).  This could all be redone in Cython anyway when time permits.

Reply:
This is simple enough that it doesn't matter that much to me whether you change it, but in general I don't think function overhead is very expensive for most languages, or at least to the point of not calling a function to save on execution time that seems like a micro optimization to me.